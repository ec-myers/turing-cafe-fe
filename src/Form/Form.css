.Form {
    align-items: center;
    display: grid;
    grid-gap: 50px;
    grid-template-columns: repeat(5, 1fr);
}

input {
  height: 25px;
  width: 200px;
}

::placeholder {
  font-size: 14px;
}

.submit-btn {
  background: #2c8080;
  border-radius: 3px;
  color: #ffffff;
  font-size: 14px;
  height: 30px;
}