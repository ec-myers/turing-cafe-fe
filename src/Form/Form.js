import React, {  Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      numGuests: ''
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }


  render() {
    return(
      <form>
        <input type='text' placeholder='Name' name='name' value={this.state.name}onChange={this.handleChange}></input>
        <input type='text' placeholder='Date' name='date' value={this.state.date} onChange={this.handleChange}></input>
        <input type='text' placeholder='Time' name='time' value={this.state.time} onChange={this.handleChange}></input>
        <input type='text' placeholder='Number of guests' name='numGuests' value={this.state.numGuests} onChange={this.handleChange}></input>
        <button type='button' onClick={this.makeReservation}>Make Reservation</button>
      </form>
    )
  }
}

export default Form;