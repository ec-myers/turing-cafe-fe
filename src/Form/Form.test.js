import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe ('Form', () => {
  it('should match the snapshot', () => {
    const mockAddReservation = jest.fn();
    let wrapper = shallow(<Form 
      addReservation={mockAddReservation} />);
    
    expect(wrapper).toMatchSnapshot();
  });
})