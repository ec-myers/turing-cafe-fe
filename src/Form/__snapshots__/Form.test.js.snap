// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="Form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date(mm/dd)"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="text"
    value={6}
  />
  <button
    className="submit-btn"
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</form>
`;
